### PR TITLE
Workspace now stores ACL group map as EnumType -> GroupRef

### PR DESCRIPTION
**Pull request by helgridly**
A few changes here:
1. Turned WorkspaceAccessLevel into a RawlsEnumeration rather than a Java Enumeration.
2. Storing ACL groups in the workspace as `Map[WorkspaceAccessLevel, RawlsGroupRef]`
3. GraphDAO can now serialize `Map[RawlsEnumeration, _]` as well as `Map[String, _]`
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [x] **Submitter**: Verify swagger UI on dev server still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by helgridly** [comment]
@dvoet In this pull request, I made `Map[RawlsEnumeration, _]` a thing we can serialize in the graph.

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala [hunk 1]
```diff
@@ -481,6 +489,16 @@ trait GraphDAO {
     }).toMap
   }
 
+  private def loadRawlsEnumMap[R <: RawlsEnumeration[R]](keyType: Type, valuesType: Type, propName: String, vertex: Vertex): Map[R, Any] = {
+    val mapDummy = getVertices(vertex, Direction.OUT, EdgeSchema.Own, propName).head
+    mapDummy.asInstanceOf[OrientVertex].getRecord.setAllowChainedAccess(false)
+
+    val boo: immutable.Map[R, Any] = getVertexKeys(mapDummy).map({ key =>
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: Is this the special Halloween edition of foo?

Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: Much better than some of the words I use for scala.

**Comment by jmthibault79** [comment]
:+1: but let's leave it open for comments from Doge & friends

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala [hunk 2]
```diff
@@ -392,7 +401,7 @@ trait GraphDAO {
       case None => throw new RawlsException("Cannot find Group corresponding to " + ref.groupName)
     }
 
-  private def saveProperty(tpe: Type, propName: String, valToSave: Any, vertex: Vertex, wsc: Option[WorkspaceContext], graph: Graph): Unit = {
+  private def saveProperty[R <: RawlsEnumeration[R]](tpe: Type, propName: String, valToSave: Any, vertex: Vertex, wsc: Option[WorkspaceContext], graph: Graph): Unit = {
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: what does this type param do?

Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: Looks like the remnants of a previous attempt to make this work. Good catch!

**Pull request merged**